src/summaries/smart_contracts.yaml

```diff
@@ -0,0 +1,159 @@
+      * These are extremely powerful because it allows users to trustlessly encode
+      rules for any type of transaction and even create scarce assets with
+      specialized functionality.
+      * Many standard business contracts can be algorithmically encoded and
+      algorithmically enforced. These contracts run on the Ethereum blockchain and are run on every node.
+      * This is not just useful for finance. For example, supply chains are another
+      good example.
+
+    - Gas 
+      * Transactions have a gas fee that is the fee received by the miners for conducting transactions.
+      * Transactions have a gas fee, which varies based on the complexity
+      of the transaction. E.g., low gas fee is used to compensate a miner
+      for including and executing a transaction, and high gas fee for
+      more data-intensive transactions.
+      * If a transaction reverts due to any failure or runs out of gas the
+      miner forfeits all gas used until that point. This helps miners earn even if large volumes of failed transactions occur.
+
+      - Gas fees
+        * The gas price is determined by the market and effectively creates
+        an auction for inclusion in the next Ethereum block.
+        * Higher gas fees signal higher demand and therefore generally
+        receive higher priority for inclusion.
+      
+      - Gas Price 
+        * Every operation on the Ethereum blockchain requires gas.
+        * Each unit of gas has a gas price, which is denoted in gwei. 1 gwei = 1
+          billionth of an ETH. Gas prices increase as users outbid one another
+          during times of higher network congestion.
+      
+      - Gas Limit
+        * The gas limit is the maximum amount of gas a user is willing to use for a
+        transaction. A transaction can include multiple operations.
+        * If there is residual amount of gas after the transaction is complete, then the user gets back the remaining gas.
+
+      - Turing Complete
+        * Gas plays a very important role. If a malicious attack is planned then it would be extremely expensive to carry out.
+        * Ethereum is turing complete whereas Bitcoin is not.
+
+    - Oracles
+      * Ethereum blockchain nows only of what happens inside the Blockchain. But how does it get the information needed from the outside world? Here Oracles come into play.
+      * An oracle, in the context of smart contract platforms, is any data source for reporting information external to the blockchain. 
+
+      - Oracle Problem
+        * This refers to the correctness of the off-chain information that is provided by the oracle.
+        * One Ethereum-based platform known as Chainlink is designed to solve the
+        oracle problem by using an aggregation of data sources. The Chainlink
+        whitepaper includes a reputation-based system that is decentralized.
+
+      
+    
+- title: Fungible and Non fungible Tokens
+  shortTitle: FT and NFTs
```

Comment:
Heading is fungible tokens and non fungible tokens, but we have not explained NFTs. We should explain that too